miner/worker.go

```diff
@@ -375,6 +379,8 @@ func (w *worker) newWorkLoop(recommit time.Duration) {
 		case head := <-w.chainHeadCh:
 			headNumber := head.Block.NumberU64()
 			clearPending(headNumber)
```

Comment:
Do you also need to add a `clearPending` line to your newView handler?  If it does, then I think you to reconsider emitting the NewView event to the worker's channel on whenever there is a new view (as opposed to only if this node is the proposer).

Reply:
Why would we emit the NewView event every time? Is it because we need to let all validators be aware of the new view/round, thus they can clearPending()?

Reply:
So i'm thinking of the case that a node has become a validator for a view and then sends out a proposal.  And then a new view happens.  What will need to happen to that in-process proposal?  I suspect that `clearPending` will need to be called, but I'm not 100% sure.  Can you look into that.

Reply:
Yes, `clearPending` will need to be called here